src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -341,6 +341,33 @@
       result
       (recur db-spec))))
 
+(defn garbage-collector
+  [db clean-lock config]
+  (let [seconds-pos? (comp pos? to-seconds)
+        what (filter identity
+                     [(when-let [node-ttl (:node-ttl config)]
```

Comment:
These nested whens can be a bit terser with `some->`, eg. `(when (some-> config :node-ttl to-seconds pos?) ...`